### PR TITLE
Update olric

### PR DESCRIPTION
**Pull request by hasit**
### Description of change
Screenshot 2023-05-24 at 19 26 45

##### Checklist

- [ ] Tested in playground or other setup
- [ ] Screenshot (Grafana) from playground added to PR for 15+ minute run
- [ ] Documentation is changed or added
- [ ] Tests and/or benchmarks are included
- [ ] Breaking changes

### Summary by OpenAI

**New Feature:**
- Updated DistCache with new client and improved methods
- Added context parameter to flow control actuators, rate limiters, and schedulers

**Refactor:**
- Modified Revert functions in quota-scheduler and workload-scheduler with context parameter

> 🎉 A cache update, swift and grand,
> With context flowing through the land.
> Actuators, limiters, schedulers unite,
> For a codebase shining bright! 🌟

### CODE REVIEW

File: pkg/rate-limiter/global-token-bucket/global-token-bucket.go [hunk 1]
```diff
@@ -122,7 +123,7 @@ func (gtb *GlobalTokenBucket) TakeIfAvailable(label string, n float64) (bool, fl
 		return true, 0, 0
 	}
 
-	resultBytes, err := gtb.dMap.Function(label, TakeNFunction, reqBytes)
+	resultBytes, err := gtb.dMap.Function(context.Background(), label, TakeNFunction, reqBytes)
 	if err != nil {
 		log.Autosample().Error().Err(err).Str("dmapName", gtb.dMap.Name()).Msg("error taking from token bucket")
 		return true, 0, 0
```
[the remark this answers is not in the document]
Comment (harjotgill) [reply pkg/rate-limiter/global-token-bucket/global-token-bucket.go:129]: @hasit we should indeed pass the context in this version

File: pkg/rate-limiter/global-token-bucket/global-token-bucket.go [hunk 2]
```diff
@@ -161,7 +162,7 @@ func (gtb *GlobalTokenBucket) Take(label string, n float64) (bool, time.Duration
 		return true, 0, 0, 0
 	}
 
-	resultBytes, err := gtb.dMap.Function(label, TakeNFunction, reqBytes)
+	resultBytes, err := gtb.dMap.Function(context.Background(), label, TakeNFunction, reqBytes)
 	if err != nil {
 		log.Autosample().Error().Err(err).Str("dmapName", gtb.dMap.Name()).Msg("error taking from token bucket")
 		return true, 0, 0, 0
```
[the remark this answers is not in the document]
Comment (hasit) [reply pkg/rate-limiter/global-token-bucket/global-token-bucket.go]: [bf77567](https://github.com/fluxninja/aperture/pull/2033/commits/bf77567ad10b29788e17935a715bfd6b1411f15c) addresses this.

File: pkg/rate-limiter/global-token-bucket/global-token-bucket.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (hasit) [reply pkg/rate-limiter/global-token-bucket/global-token-bucket.go]: [bf77567](https://github.com/fluxninja/aperture/pull/2033/commits/bf77567ad10b29788e17935a715bfd6b1411f15c) addresses this.

**Comment by hasit** [comment]
I see inaccurate results in tests. I am trying to figure out what's causing that.

**Pull request closed without merging**